src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java

```diff
@@ -578,6 +601,79 @@ private boolean createComponent(Channel channel, User sender, String subcomponen
         return result;
     }
 
+    String createUploadPermissionPR(Channel channel, User sender, String jiraIssue, String forkTo, List<String> ghUsers, List<String> releaseUsers) {
+        String prUrl = "";
+        boolean isPlugin = forkTo.endsWith("-plugin");
+        OutputChannel out = channel.send();
+        if(isPlugin) {
+            String branchName = "ircbot-" + forkTo + "-permissions";
+            try {
+                if (releaseUsers.isEmpty()) {
+                    out.message("No users defined for release permissions, will not create PR");
+                    return null;
+                }
+
+                GitHub github = GitHub.connect();
+                GHOrganization org = github.getOrganization(IrcBotConfig.GITHUB_INFRA_ORGANIZATION);
+                GHRepository repo = org.getRepository(IrcBotConfig.GITHUB_UPLOAD_PERMISSIONS_REPO);
+
+
+                GHContent prTemplate = repo.getFileContent(".github/PULL_REQUEST_TEMPLATE.md");
+                if (prTemplate == null) {
+                    out.message("Could not retrieve PR template for " + IrcBotConfig.GITHUB_UPLOAD_PERMISSIONS_REPO);
+                    return null;
+                }
+
+                String artifactPath = getArtifactPath(channel, github, forkTo);
+                if (StringUtils.isBlank(artifactPath)) {
+                    out.message("Could not resolve artifact path for " + forkTo);
+                    return null;
+                }
+
+                if(!repo.getBranches().containsKey(branchName)) {
+                    repo.createRef("refs/heads/" + branchName, repo.getBranch("master").getSHA1());
+                }
+
+                GHContentBuilder builder = repo.createContent();
+                builder = builder.branch(branchName).message("Adding upload permissions file for " + forkTo);
+                String name = forkTo.replace("-plugin", "");
+                String content = "---\n" +
+                        "name: \"" + name + "\"\n" +
+                        "github: \"jenkinsci/" + forkTo + "\"\n" +
+                        "paths:\n" +
+                        "- \"" + artifactPath + "\"\n" +
+                        "developers:\n";
+                StringBuffer userBuffer = new StringBuffer();
```

Comment:
you almost never want to use stringbuffer, it's synchronised.
```suggestion
                StringBuilder developerBuilder = new StringBuilder();
```